Update README.md (#3)
SiriosDev/SwitchBot-API-Script-Caller: README.md

## Requirements
- HACS ([doc](https://hacs.xyz/docs/setup/prerequisites))
    - PyScript Integration ([doc](https://hacs-pyscript.readthedocs.io/en/latest/installation.html)) with `import-all` option activated
      

## Installation

> ### Procedure

Reviewer: I like the procedure, but I have some important changes to make, so if it's not a problem for you, I will make some change after the merge

   ```yaml
   # configuration.yaml
   # (...)
   pyscript: !include deps/SwitchBot-API-Script-Caller/pyscript/config.yaml
   # (...)
   ```
5. **Set the authentication secrets in `secrets.yaml` homeassistant file**
    - Random Value (`switchbot_nonc`) (I suggest using an UUID generator, but any unique alphanumeric string is fine)
    ```yaml
    # secrets.yaml
    # (...)
    # Token and Secret Key : from `Developer Option` in the SwitchBot App (version ≥6.14)
    switchbot_token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    switchbot_sec: xxxxxxxxxxxx
    # Random Value: you can use a UUID generator, but any unique alphanumeric string is OK
    switchbot_nonc: xxxxx-xxxxx-xxxxx-xxxxx-xxxxx
    ```
6. **Link the files in the `pyscript` directory**
   ```sh
   # use `mkdir -p pyscript/apps/` if the directory doesn't exist
   cd pyscript/apps/
   
   # Create a symbolic link to the apps directory named switchbot
   ln -s ../../deps/SwitchBot-API-Script-Caller/pyscript/apps switchbot
   ```
   
### Further Update
By following this procedure, the script can then be updated with newer version using git.
```sh
cd deps/SwitchBot-API-Script-Caller
git pull
```

### Installation Notes

> - In order to see the `Developper options` in the Switchbot app, click repetively on the version number in the App's settings

Reviewer: add minumum version of app as v.6.14 [here](https://github.com/OpenWonderLabs/SwitchBotAPI#open-token-and-secret-key)

Author: ```suggestion
- In order to see the `Developper options` in the Switchbot app (version ≥6.14), click repetively on the version number in the App's settings
```

- A symbolic link is symbolic and represent the exact path you enter, if you move the targeted file or if the target is outside of the container (e.g. when using docker) the link will not work. Make sure that you are using a relative path that is accessible for the host reading the link. 
- Ensure that `pyscript` is operational before to install this script.

## How To Use
This script (for now) provides two services in home assisant:

### Summary
- [SwitchBot HVAC API Interface (`pyscript.switchbot_hvac`)](#switchbot-hvac-api-interface)
- [SwitchBot Generic Command API Interface (`pyscript.switchbot_generic_command`)](#switchbot-generic-command-api-interface)

### SwitchBot HVAC API Interface

> _Interface for HVAC (heating, ventilation and air conditioning) device._

Reviewer: I think it is better to specify here that it has to be an HVAC through IR control

Author: ```suggestion
_Interface for infrared HVAC (heating, ventilation and air conditioning) device._
```

    - string value between `on` and `off`

### SwitchBot Generic Command API Interface
_Allows you to send any request to the API. (See [documentation][generic-cmd-link])_

**Parameters:**
- `deviceId:`
    - to get this id read [here][deviceid-link]
- `command:`

>     - One of the command supported by the device. (see [documentation](https://github.com/OpenWonderLabs/SwitchBotAPI#command-set-for-virtual-infrared-remote-devices))

Reviewer: I think it's better to put the generic-cmd-link here because they have the same fields, the only difference between the normal and ir commands are the ids

Author: ```suggestion
    - One of the command supported by the device. (see [documentation][generic-cmd-link])
```
